### DISCUSSION TITLE
tabbed, fullscreen, scratchpad, context, state save/load...

### DISCUSSION THREAD

**Discussion by yuuyins** [MagmaWM/MagmaWM#49] [Ideas]
- tabbed windows layout
- toggle window to fullscreen
- scratchpads (which we can set to floating)
- space contexts akin to https://github.com/procrat/xmonad-contexts
- state persistence (saving/loading contexts/workspaces, akin to https://github.com/overideal/perject)

**Comment by electron271** [comment]
fullscreen will be implemented and scratchpads we are looking into

state persistence might be a bit harder but ill take a look

**Reply by yuuyins** [reply]
Thank you! And what is your perspective on contexts and tabbed? Those are also a primary requirement for me specifically. As I learn Rust more, I would like contributing if I'm able to.

Really looking forward into MagmaWM as a replacement for my current Xmonad setup, as MagmaWM also seems to target extensibility through packages, right?

**Reply by electron271** [reply]
tabbed layouts seems like a cool idea, but not exactly a priority with how much work things need rn :P 
and contexts seem quite interesting and ill look later at the github repo you sent

**Reply by electron271** [reply]
could you open a issue for this so we can track it easier?
